Source/ASCollectionNode.mm

```diff
@@ -275,6 +277,44 @@ - (BOOL)inverted
   }
 }
 
+- (void)setLayoutInspector:(id<ASCollectionViewLayoutInspecting>)layoutInspector
+{
+  if ([self pendingState]) {
+    _pendingState.layoutInspector = layoutInspector;
+  } else {
+    ASDisplayNodeAssert([self isNodeLoaded], @"ASCollectionNode should be loaded if pendingState doesn't exist");
+    self.view.layoutInspector = layoutInspector;
+  }
+}
+
+- (id<ASCollectionViewLayoutInspecting>)layoutInspector
+{
+  if ([self pendingState]) {
+    return _pendingState.layoutInspector;
+  } else {
+    return self.view.layoutInspector;
+  }
+}
+
+- (void)setLeadingScreensForBatching:(CGFloat)leadingScreensForBatching
+{
+  if ([self pendingState]) {
+    _pendingState.leadingScreensForBatching = leadingScreensForBatching;
+  } else {
+    ASDisplayNodeAssert([self isNodeLoaded], @"ASCollectionNode should be loaded if pendingState doesn't exist");
+    self.view.leadingScreensForBatching = leadingScreensForBatching;
+  }
+}
+
+- (CGFloat)leadingScreensForBatching
+{
+  if ([self pendingState]) {
+    return _pendingState.leadingScreensForBatching;
+  } else {
+    return self.view.leadingScreensForBatching;
+  }
+}
+
```

Comment:
I guess the pending state would be another improvement we could build some kind of broader infrastructure around it.